modules/core/src/main/java/org/apache/ignite/internal/processors/service/IgniteServiceProcessor.java

```diff
@@ -638,7 +642,7 @@ private void ensure(boolean cond, String desc, @Nullable Object v) {
      */
     private PreparedConfigurations<IgniteUuid> prepareServiceConfigurations(Collection<ServiceConfiguration> cfgs,
         IgnitePredicate<ClusterNode> dfltNodeFilter) {
-        List<ServiceConfiguration> cfgsCp = new ArrayList<>(cfgs.size());
+        List<LazyServiceConfiguration> cfgsCp = new ArrayList<>(cfgs.size());
```

Comment:
In most cases there is no need to change `ServiceConfiguration` to `LazyServiceConfiguration`. The latter inherits from the former, so we can safely use `ServiceConfiguration` everywhere and cast it to `Lazy...` in single place only.

Reply:
Fixed